server/test/repository/VersionRepositoryTest.java

```diff
@@ -39,17 +39,57 @@ public void setupVersionRepository() {
 
   @Test
   public void testPublish() {
-    resourceCreator.insertActiveProgram("foo");
-    resourceCreator.insertActiveProgram("bar");
-    resourceCreator.insertDraftProgram("bar");
-    assertThat(versionRepository.getActiveVersion().getPrograms()).hasSize(2);
-    assertThat(versionRepository.getDraftVersion().getPrograms()).hasSize(1);
+    Question firstQuestion = resourceCreator.insertQuestion("first-question");
+    firstQuestion.addVersion(versionRepository.getActiveVersion()).save();
+    Question secondQuestion = resourceCreator.insertQuestion("second-question");
+    secondQuestion.addVersion(versionRepository.getActiveVersion()).save();
+
+    Program firstProgramActive =
+        ProgramBuilder.newActiveProgram("foo")
+            .withBlock("Screen 1")
+            .withRequiredQuestion(firstQuestion)
+            .build();
+    Program secondProgramActive =
+        ProgramBuilder.newActiveProgram("bar")
+            .withBlock("Screen 1")
+            .withRequiredQuestion(secondQuestion)
+            .build();
+    Question secondQuestionUpdated = resourceCreator.insertQuestion("second-question");
```

Comment:
nit: Updated -> Draft for parity with the Draft program that immediately follow

Reply:
Done!